test/provisioner/provision.go

```diff
@@ -121,6 +222,177 @@ func GetInstallOverlay(provider string, installDir string) (InstallOverlay, erro
+	}, nil
+}
+
+func (lio *KbsInstallOverlay) Apply(ctx context.Context, cfg *envconf.Config) error {
+	return lio.overlay.Apply(ctx, cfg)
+}
+
+func (lio *KbsInstallOverlay) Delete(ctx context.Context, cfg *envconf.Config) error {
+	return lio.overlay.Delete(ctx, cfg)
+}
+
+func (lio *KbsInstallOverlay) Edit(ctx context.Context, cfg *envconf.Config, props map[string]string) error {
+	var err error
+	log.Infof("Updating kbs image with %q", props["KBS_IMAGE"])
+	if err = lio.overlay.SetKustomizeImage("kbs-container-image", "newName", props["KBS_IMAGE"]); err != nil {
+		return err
+	}
+
+	log.Infof("Updating CAA image tag with %q", props["KBS_IMAGE_TAG"])
+	if err = lio.overlay.SetKustomizeImage("kbs-container-image", "newTag", props["KBS_IMAGE_TAG"]); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (p *KeyBrokerService) GetKbsSvcIP(ctx context.Context, cfg *envconf.Config) (string, error) {
+	client, err := cfg.NewClient()
+	if err != nil {
+		return "", err
+	}
+
+	namespace := "coco-tenant"
+	deploymentName := "kbs"
+
+	err = AllPodsRunning(ctx, cfg, namespace)
+	if err != nil {
+		err = fmt.Errorf("All pods are not running: %w\n", err)
+		log.Errorf("%v", err)
+		return "", err
+	}
+
+	resources := client.Resources(namespace)
+
+	// Get the service associated with the deployment
+	serviceList := &corev1.ServiceList{}
+	err = resources.List(context.TODO(), serviceList)
+	if err != nil {
+		err = fmt.Errorf("Error listing services: %w\n", err)
+		log.Errorf("%v", err)
+		return "", err
+	}
+
+	var matchingService *corev1.Service
+	for i := range serviceList.Items {
+		service := &serviceList.Items[i]
+		if service.Name == deploymentName {
+			matchingService = service
+			break
+		}
+	}
+
+	if matchingService == nil {
+		return "", fmt.Errorf("No service with label selector found")
+	}
+
+	fmt.Printf("Service IP: %s\n", matchingService.Spec.ClusterIP)
+	return matchingService.Spec.ClusterIP, nil
+}
+
+func (p *KeyBrokerService) Deploy(ctx context.Context, cfg *envconf.Config, props map[string]string) error {
+	log.Info("Customize the overlay yaml file")
+	if err := p.installOverlay.Edit(ctx, cfg, props); err != nil {
+		return err
+	}
+
+	originalDir, err := os.Getwd()
+	if err != nil {
+		err = fmt.Errorf("Error getting the current working directory: %w\n", err)
+		log.Errorf("%v", err)
+		return err
+	}
+
+	newDirectory := "kbs/kbs/config/kubernetes"
+	err = os.Chdir(newDirectory)
+	if err != nil {
+		err = fmt.Errorf("Error changing the working directory: %w\n", err)
+		log.Errorf("%v", err)
+		return err
+	}
+
+	// Replace this to use install overlay
+	cmd := exec.Command("kubectl", "apply", "-k", "overlays")
+	cmd.Env = append(os.Environ(), fmt.Sprintf("KUBECONFIG="+cfg.KubeconfigFile()))
+	stdoutStderr, err := cmd.CombinedOutput()
+	log.Tracef("%v, output: %s", cmd, stdoutStderr)
+	if err != nil {
+		return err
+	}
+
+	/*
+		log.Info("Install Kbs")
+		if err := p.installOverlay.Apply(ctx, cfg); err != nil {
+			return err
+		}
+	*/
```

Comment:
leftover?

Reply:
Keeping it to make changes to way to apply the overlay yaml.